ci/scripts/r_test.sh

```diff
@@ -90,11 +90,23 @@ SCRIPT="as_cran <- !identical(tolower(Sys.getenv('NOT_CRAN')), 'true')
       message('Running minio for S3 tests (if build supports them)')
       minio_dir <- tempfile()
       dir.create(minio_dir)
-      pid <- sys::exec_background('minio', c('server', minio_dir))
-      on.exit(tools::pskill(pid))
+      pid_minio <- sys::exec_background('minio', c('server', minio_dir))
+      on.exit(tools::pskill(pid_minio), add = TRUE)
     }
   }
 
+  if(reticulate::py_module_available('pyarrow')){
+    message('Running flight demo server for tests.')
+    pid_flight <- sys::exec_background(
+      'R', 
+      c(
+        '-e',
+        'reticulate::import_from_path(\"demo_flight_server\", \"./inst\")\$DemoFlightServer(port = 8089)\$serve()'
```

Comment:
Is there a reason why we are going through `reticulate` instead of just invoking python to start the server?

Reply:
I did it in R to use the same method as was used for minio, we could move both of these outside of the R script if that makes more sense for you?